File: scripts/algorithms/Reverse String/Reverse String.js
var reverseString = function(s) {
    for(let i = 0 ; i < s.length / 2 ; i++){
        [s[i], s[s.length - i - 1]] = [s[s.length - i - 1], s[i]]
    }
    return s;
};

File: scripts/algorithms/R/Reverse String/Reverse String.cpp
// Runtime: 11 ms (Top 95.78%) | Memory: 23.70 MB (Top 32.89%)

class Solution {
public:
    void reverseString(vector<char>& s) {
        int i = 0;
        int j = s.size() - 1;
        
        while(i <= j){
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            i++;
            j--;
        }
    }
};
